### ISSUE TITLE
Relative and absolute preset paths don't work

### ISSUE THREAD

**Issue by DominicTobias** [babel/babel#8734]
## Bug Report

**Current Behavior**
Cannot find the plugin.

```js
const babelCore = require('@babel/core');

babelCore.transform('code', {
  presets: ['/Users/infensus/Sites/yebu/yebu-cli/node_modules/@babel/preset-es2015'],
});
```

I've also tried a relative path (i.e. `../yebu-cli/node_modules/@babel/preset-es2015'.

The path is definitely correct:

![screen shot 2018-09-19 at 20 24 36](https://user-images.githubusercontent.com/760314/45776353-846ae580-bc4a-11e8-9645-81faec916d4a.png)

**Expected behavior/code**
Finds the plugin by absolute or relative path.

**Environment**
- Babel version(s): 7.1.0
- Node/npm version: 10.10
- OS: MacOS 10.13.5

**Comment by nicolo-ribaudo** [comment]
You need to add the `module:` prefix. e.g.
```js
babelCore.transform('code', {
  presets: ['module:/Users/infensus/Sites/yebu/yebu-cli/node_modules/@babel/preset-es2015'],
});
```

Also, since you are using JS files you could just require the preset:
```js
babelCore.transform('code', {
  presets: [
    require('/Users/infensus/Sites/yebu/yebu-cli/node_modules/@babel/preset-es2015')
  ],
});
```

**Comment by loganfsmyth** [comment]
Hmm, `module:` would indeed make it work for now, but I agree that it shouldn't be needed ideally. If it's not working, that's a mistake. We can easily see that the path is absolute or starts with `./` or `../`.

**Comment by loganfsmyth** [comment]
I'm actually not able to reproduce this. Would you be able to create a reproduction example?

**Comment by DominicTobias** [comment]
Thanks for trying, I will put together an example later.

I think the problem must be around:

Folder A is a CLI tool with the code and babel.

It’s executed inside Folder B. If I start to follow through the preset
resolve functions the dirname param is that of Folder B.

I think it must be something to do with executing the code in the other
directory.

**Comment by fangyongbao** [comment]
I also encountered this problem, I hope the authorities can fix it as soon as possible

**Comment by fangyongbao** [comment]
![image](https://user-images.githubusercontent.com/6182610/45816339-e17b9f80-bd0d-11e8-9374-091771197798.png)
Execute the cli command in the project directory. An error (Couldn 't find preset "es2015" to the directory "/ Users/yongbao fyb/Desktop).

**Comment by DominicTobias** [comment]
I’m going to make a MVP later and also try and trace it through and find
the route cause.

**Comment by DominicTobias** [comment]
This is working for me when I changed from `@babel/preset-es2015` to `@babel/preset-env`.

Only thing I'd mention is that the error could be more helpful which was:

> Error: Cannot find module '/Users/infensus/Sites/yebu/yebu-cli/node_modules/@babel/preset-es2015' from '/Users/infensus/Sites/yebu/test-app'

Rather than saying it was incompatible. Unless it was complaining because preset-2015 doesn't have anything in it except a package.json with a main entry pointing at a non-existant file or at best /node_modules if there are conflicting versions (it's on in the screenshot because I installed preset-env which had some conflicting versions).

![screen shot 2018-09-20 at 19 09 26](https://user-images.githubusercontent.com/760314/45838143-f3a70f00-bd08-11e8-9ce1-bca60bace58c.png)
